src/opera/pge/disp_s1/disp_s1_pge.py

```diff
@@ -139,6 +142,188 @@ def _validate_output(self):
+
+        <CoreFilename>_<Mode>_<FrameID>_<Polarization>_<ReferenceDateTime>_<SecondaryDateTime>_<ProductVersion>_<ProductGenerationDateTime>
+
+        Where <CoreFilename> is returned by DispS1PostProcessorMixin._core_filename()
+
+        Callers of this function are responsible for assignment of any other
+        product-specific fields, such as the file extension.
+
+        Parameters
+        ----------
+        inter_filename : str, optional
+            The intermediate filename of the output product to generate the
+            core filename for. This parameter may be used to inspect the file
+            in order to derive any necessary components of the returned filename.
+            Once the core filename is cached upon first call to this function,
+            this parameter may be omitted.
+
+        Returns
+        -------
+        frame_filename : str
+            The filename component to assign to frame-based products created by
+            this PGE.
+        """
+        core_filename = self._core_filename(inter_filename)
+
+        disp_metadata = self._collect_disp_s1_product_metadata(inter_filename)
+
+        # Mode: S1-A/B acquisition mode, should be fixed to "IW"
+        mode = "IW"
+
+        # FrameID: Unique frame identification number as a 5-digit string in
+        # the format FXXXXX
+        frame_id = f"F{disp_metadata['identification']['frame_id']:05d}"
+
+        # Polarization:  polarization of the input bursts
+        # derived from names of input CSLCs in the runconfig
+        cslc_file_list = self.runconfig.sas_config['input_file_group']['cslc_file_list']
+        pattern = re.compile(r"t\w{3}_\d{6}_iw[1|2|3]_[0-9]{8}_(VV|VH).h5")
+        for cslc_file in cslc_file_list:
+            cslc_file_basename = basename(cslc_file)
+            if pattern.match(cslc_file_basename):
+                polarization = cslc_file_basename.split('_')[-1].split('.')[0]
+                break
+        else:
+            raise RuntimeError(
+                'No cslc_file_list file matches the expected naming pattern to '
+                'retrieve the polarization information.'
+            )
+
+        # ReferenceDateTime: The acquisition sensing start date and time of
+        # the input satellite imagery for the first burst in the frame of the
+        # reference product in the format YYYYMMDDTHHMMSSZ
+        # TODO This should come from the product metadata under
+        # "identification/reference_datetime", however, it is missing from
+        # the current delivery. We should assign a placeholder datetime in
+        # the meantime
+        reference_date_time = "YYYYMMDDTHHMMSSZ"
+
+        # SecondaryDateTime: The acquisition sensing start date and time of
+        # the input satellite imagery for the first burst in the frame of this
+        # secondary product in the format YYYYMMDDTHHMMSSZ
+        # TODO This should come from the product metadata
+        secondary_date_time = "YYYYMMDDTHHMMSSZ"
```

Comment:
Rather than define the placeholders here, I would prefer to see them in `_collect_disp_s1_product_metadata`, since that has been the traditional place where the PGE makes modifications to the metadata dictionary to make up for shortfalls in the current product metadata.

Also, rather than the literal "YYYYMMDDTHHMMSSZ", we should use an actual datetime string since PCM will be expecting an actual datetime to pattern match against. The actual dates themselves can be completely made up so its obvious they are placeholders, such as "20230101T000000Z"